Library/Homebrew/dependency.rb

```diff
@@ -71,25 +71,30 @@ class << self
     # The default filter, which is applied when a block is not given, omits
     # optionals and recommendeds based on what the dependent has asked for.
     def expand(dependent, deps = dependent.deps, &block)
+      @expand_stack ||= []
```

Comment:
Just one more thing: I'd like to see a comment above that line that states something along the lines of “Keep track of already expanded/visited dependencies to avoid infinite recursion with cyclic dependencies.”